File: solutions/263/263.ugly-number.php
/*
 * @lc app=leetcode id=263 lang=php
 *
 * [263] Ugly Number
 */

// @lc code=start
class Solution {

    /**
     * @param Integer $n
     * @return Boolean
     */
    function isUgly($n) {
        if ($n < 1) {
            return false;
        }

        while ($n % 2 === 0) {
            $n = $n / 2;
        }
        while ($n % 3 === 0) {
            $n = $n / 3;
        }
        while ($n % 5 === 0) {
            $n = $n / 5;
        }

        return $n === 1;
    }
}
// @lc code=end

File: solutions/263/263.ugly-number.go
/*
 * @lc app=leetcode id=263 lang=golang
 *
 * [263] Ugly Number
 */

// @lc code=start
func isUgly(n int) bool {
    if n <= 0 {
		return false
	}

	for _, v := range [3]int{2 ,3 ,5} {
		for n % v == 0 {
			n /= v
		}
	}

	return n == 1
}
// @lc code=end

